From f1158b32d3c8874536efcf7dc3f4f3050d976b25 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Thu, 8 Oct 2020 16:35:39 +0200
Subject: [PATCH] BusyIndicator: Do not rotate when invisible

It seems like the "visible" property is catching there is the wrong one
and would still rotate while invisible.

CCBUG: 426746
(cherry picked from commit e0e7840a0307deaee7a0b63f341973f8e3508873)
---
 src/declarativeimports/plasmacomponents3/BusyIndicator.qml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/declarativeimports/plasmacomponents3/BusyIndicator.qml b/src/declarativeimports/plasmacomponents3/BusyIndicator.qml
index d081fc43d..ec4ccaf13 100644
--- a/src/declarativeimports/plasmacomponents3/BusyIndicator.qml
+++ b/src/declarativeimports/plasmacomponents3/BusyIndicator.qml
@@ -25,6 +25,7 @@ T.BusyIndicator {
     padding: 0
 
     contentItem: Item {
+        id: baseItem
         /* implicitWidth and implicitHeight won't work unless they come
          * from a child of the contentItem. No idea why.
          */
@@ -61,7 +62,7 @@ T.BusyIndicator {
                 to: 360
                 duration: 2000
                 loops: Animation.Infinite
-                running: visible
+                running: control.running || baseItem.opacity > 0
             }
         }
     }
-- 
2.25.1

